pkg/k8sutil/resource.go

```diff
@@ -33,6 +33,41 @@ import (
 	runtimeClient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+func Delete(log logr.Logger, client runtimeClient.Client, target runtime.Object) error {
+	targetType := reflect.TypeOf(target)
+	current := target.DeepCopyObject()
+
+	key, err := runtimeClient.ObjectKeyFromObject(current)
+	if err != nil {
+		return errors.WithDetails(err, "kind", targetType)
+	}
+	log = log.WithValues("kind", targetType, "name", key.Name)
+
+	err = client.Get(context.TODO(), key, current)
+	if err == nil {
+		err = client.Delete(context.TODO(), current)
+		if err != nil {
+			return errorfactory.New(
+				errorfactory.APIFailure{},
+				err,
+				"delete resource failed",
+				"kind", targetType, "name", key.Name,
+			)
+		}
+	} else if apierrors.IsNotFound(err) {
+		return nil
```

Comment:
do we need at least log a warn message here?

Reply:
I thought about this, but I'm not sure. This might get pretty noise. Every time we reconcile the resources that can be either global or group-based (e.g.: Envoy ingress deployment), we reconcile the global and delete the group-based or the other way around (depending on the config). So, it might get noisy and confusing. wdyt?

Reply:
Can't we log this on at least on `DEBUG` level?